### PR TITLE
Feature/add week get service for user contract

### PR DESCRIPTION
**Pull request by Rapkup**
Added:
-BulkWeekOfEmployeeUserContracts endpoint & service
-GetAllEmployeeContracts service
-RemoveAllEmployeeContractsAsync endpoint & service

### CODE REVIEW

**Review by IvanLukyanau — CHANGES_REQUESTED** [review]
Checkout develop, pull, checkout 'feature/add-week-get-service-for-user-contract', merge develop into current

**Comment by Rapkup** [comment]
> Checkout develop, pull, checkout 'feature/add-week-get-service-for-user-contract', merge develop into current

Done!

**Review by IvanLukyanau — CHANGES_REQUESTED** [review]
Please fix all comments

File: LX.StaffScheduler.BLL/Services/Common/UserContract.cs [hunk 1]
```diff
@@ -22,18 +22,56 @@ public async Task<UserContractDTO> AddAsync(UserContractDTO entity)
...
+        }
+
         public async Task<List<UserContractDTO>> GetAllAsync()
         {
             var userContracts = await repository.GetAllAsync();
             return userContracts.Select(userContract => userContract.UserContractToDTO()).ToList();
         }
+       public async Task<IEnumerable<UserContractDTO>> GetAllEmployeeContracts(int userId)
+        {
+            var allContracts = await repository.GetAllAsync();
```
[the remark this answers is not in the document]
Comment (Rapkup) [reply LX.StaffScheduler.BLL/Services/Common/UserContract.cs:47]: Logic of this mrthod removed to DAL layer

File: LX.StaffScheduler.BLL/Services/Common/UserContract.cs [hunk 2]
```diff
@@ -22,18 +22,56 @@ public async Task<UserContractDTO> AddAsync(UserContractDTO entity)
...
 
         public async Task<UserContractDTO?> GetByIdAsync(int id)
         {
             var userContract = await repository.GetByIdAsync(id);
             return userContract?.UserContractToDTO();
         }
 
+      public async Task RemoveAllEmployeeContractsAsync(int userId)
+        {
+            var allContracts = await repository.GetAllAsync();
```
[the remark this answers is not in the document]
Comment (Rapkup) [reply LX.StaffScheduler.BLL/Services/Common/UserContract.cs:68]: Logic of this mrthod removed to DAL layer

File: LX.StaffScheduler.Api/Controllers/UserContractsController.cs [hunk 3]
```diff
@@ -94,5 +94,35 @@ public async Task<IActionResult> Delete(int id)
...
+            }
+        }
+
+        [HttpGet("weekly-contracts/{id}")]
+        public async Task<IActionResult> GetContractsWeekSchedule(int userId)
+        {
+            try
+            {
+                _svc.GetAllEmployeeContracts(userId);
+                return Ok();
```
[the remark this answers is not in the document]
Comment (Rapkup) [reply LX.StaffScheduler.Api/Controllers/UserContractsController.cs:118]: Fixed

**Review by blattlause — APPROVED** [review]

**Review by blattlause — COMMENTED** [review]
delicious stuff man

**Review by IvanLukyanau — APPROVED** [review]

**Pull request merged**